web/src/presence.ts

```diff
@@ -297,7 +297,7 @@ export function last_active_date(user_id: number): Date | undefined {
 export function initialize(params: {
     presences: Record<number, Omit<RawPresence, "server_timestamp">>;
     server_timestamp: number;
-    presence_last_update_id: number;
+    presence_last_update_id: number | undefined;
 }): void {
     set_info(params.presences, params.server_timestamp, params.presence_last_update_id);
```

Comment:
I'm not sure this is the best way for the client to handle the spectators case; passing `params.presence_last_update_id ?? -1`  here is probably nicer for allowing presence code to have an actual value with the expected type at all times.

It's undefined for spectators because they didn't ask for a value for it, but if you didn't request presence previously, the correct value for the client-side state machine is `-1`, I think.